### DISCUSSION TITLE
Let's talk about Wlang

### DISCUSSION THREAD

**Discussion by HashLDash** [HashLDash/Photon#5] [General]
Here is where we can talk about the language and the future of the language. What you think, features, etc. Relevant discussions can become issues that will be considered in the development of the language.

**Comment by zanderlewis** [comment]
I am creating a programming language that embeds other languages into it using a [plang] tag. I was wondering if there is a way to embed it. I am creating the language in python and I want to know if there is anyway to run code directly through the command line. This is what I do for Ruby and F#. Thanks -WolfTheDeveloper

**Reply by zanderlewis** [reply]
If you can run it in python as a runtime, that would be even better.

**Reply by HashLDash** [reply]
Hello! I don't know if I understood you correctly.
Running code directly through the command line would be something like:
photon -c "print('Hello World')"      ??
You can run photon code in python, that's how the interpreter mode works. You can check that out. It wont be fast though.
You can also compile code and run that inside python using FFI.
Let me know if that helped.

**Reply by zanderlewis** [reply]
Is there any way i can embed it? For example, in my language, I would use 

```
[plang lua]
print("Hello from Lua!")
[endplang]
```

and I use the `lupa` library in the language to use the Lua runtime. If you could explain how I might achieve something similar to this or if I would just have to run code using `photon -c some-code` to achieve a link to my language? Thanks -WolfTheDeveloper

**Reply by HashLDash** [reply]
You can take a look at method run of interpreter.py
It's an example of how to execute photon code in Python.
Code is read with self.input, parsed and tokenized, then processed by the engine.
Variables can be inspected from engine.globals
Hope this helps

**Reply by HashLDash** [reply]
With the current implementation (0.0.7), the python transpiler might not work. You can use the previous version of Photon 0.0.6.
